src/common/types/types.cpp

```diff
@@ -122,12 +122,12 @@ struct_field_idx_t StructType::getFieldIdx(const LogicalType& type, const std::s
 
 LogicalType MapType::getKeyType(const LogicalType& type) {
     KU_ASSERT(type.getLogicalTypeID() == LogicalTypeID::MAP);
-    return StructType::getFieldTypes(ListType::getChildType(type))[0];
+    return std::move(StructType::getFieldTypes(ListType::getChildType(type))[0]);
```

Comment:
hmmm I'm pretty sure move a child type out of a LogicalType object can be problematic. Can we return const& here?

Reply:
functions of the form getFieldTypes/getChildType/getField(...) tend to return a copy of their child type. I'm pretty sure it's been this way since before the refactor, but I do think it should be change. I'll change the whole thing in this PR.